packages/access-client/src/space.js

```diff
@@ -0,0 +1,236 @@
+import * as ED25519 from '@ucanto/principal/ed25519'
+import { delegate, Schema, UCAN } from '@ucanto/core'
+import * as BIP39 from '@scure/bip39'
+import { wordlist } from '@scure/bip39/wordlists/english'
+import * as API from './types.js'
+import * as Access from './access.js'
+
+/**
+ * @typedef {object} Model
+ * @property {ED25519.EdSigner} signer
+ * @property {string} name
+ */
+
+/**
+ * Generates a new space.
+ *
+ * @param {object} options
+ * @param {string} options.name
+ */
+export const generate = async ({ name }) => {
+  const { signer } = await ED25519.generate()
+
+  return new OwnedSpace({ signer, name })
+}
+
+/**
+ * Recovers space from the mnemonic.
+ *
+ * @param {string} mnemonic
+ * @param {object} options
+ * @param {string} options.name
+ */
+export const fromMnemonic = async (mnemonic, { name }) => {
+  const secret = BIP39.mnemonicToEntropy(mnemonic, wordlist)
+  const signer = await ED25519.derive(secret)
+  return new OwnedSpace({ signer, name })
+}
+
+/**
+ * @param {object} space
+ * @param {ED25519.EdSigner} space.signer
+ */
+export const toMnemonic = ({ signer }) => {
+  /** @type {Uint8Array} */
+  // @ts-expect-error - Field is defined but not in the interface
```

Comment:
that's actually a little terrifying :P
usually I'd pass a `signer` instead of a `secret` and expect the `signer` is hiding any secrets and only eposing the ability to sign. good to realize that `signer` is not actually hiding that prop

Reply:
Secret is just byte range within the signer byte array it's something like `signer.subarray(3, 3 + 32)` if I'm not wrong

Reply:
See https://github.com/web3-storage/ucanto/blob/43ea497ca2200f9e1e15dbbdd033ec24895edb7d/packages/principal/src/ed25519/signer.js#L155-L164